patroni/dcs/kubernetes.py

```diff
@@ -161,30 +166,32 @@ def headers(self):
 
 class K8sObject(object):
 
-    def __init__(self, kwargs):
+    def __init__(self, kwargs: Dict[str, Any]):
         self._dict = {k: self._wrap(k, v) for k, v in kwargs.items()}
 
-    def get(self, name, default=None):
+    def get(self, name: str, default: Optional[Any] = None) -> Any:
         return self._dict.get(name, default)
 
-    def __getattr__(self, name):
+    def __getattr__(self, name: str) -> Any:
```

Comment:
```suggestion
    def __getattr__(self, name: str) -> Optional[Any]:
```

Same thoughts as the previous.

Reply:
This one will indeed create a lot of issues because not all fields in K8s objects are present in JSON.
Example:
```json
{
    "metadata": {
        "resourceVersion": "123"
        "annotations": {}
    }
}
```

In this specific case K8s API sends it as:
```json

{
    "metadata": {
        "resourceVersion": "123"
    }
}
```

We could do some workarounds like memorizing the name of the parent (in this case "metadata") and make some decisions based on it. E.g., raise AttributeError if key isn't present in the `self._dict` with exception if `metadata.annotations`, `subsets` of the endpoint are accessed.

The list of exceptions shouldn't be big, but knowing the speed of K8s development I would not rely on it.
Hence, probably we should live `__getattr__` as it is, returning Any.